library/std/src/f32/soft/tests.rs

```diff
@@ -0,0 +1,49 @@
+use crate::f32::soft::{Positive, PositiveFinite, Representation, Sign, pow2};
+
+#[test]
+fn test_representation() {
+    assert_eq!(Representation::new(-1.5f32), Representation {
+        sign: Sign::Negative,
+        abs: Positive::Finite(PositiveFinite { exp: -23, mantissa: 0xc00000 }),
+    });
+
+    assert_eq!(Representation::new(f32::MIN_POSITIVE), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: -149, mantissa: 0x800000 }),
+    });
+
+    assert_eq!(Representation::new(f32::MIN_POSITIVE / 2.0), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: -150, mantissa: 0x800000 }),
+    });
+
+    assert_eq!(Representation::new(f32::MAX), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: 104, mantissa: 0xffffff }),
```

Comment:
This is a different way to write these mantissa values than is done elsewhere...?